code/datums/components/aiming.dm

```diff
@@ -0,0 +1,165 @@
+	playsound(target, 'sound/weapons/autoguninsert.ogg', 100, TRUE)
+	new /obj/effect/temp_visual/aiming(get_turf(target))
+
+	// Register signals to alert our user if the target does something shifty.
+	RegisterSignal(src.target, COMSIG_ITEM_DROPPED, .proc/on_drop)
+	RegisterSignal(src.target, COMSIG_ITEM_EQUIPPED, .proc/on_equip)
+	RegisterSignal(src.target, COMSIG_LIVING_STATUS_PARALYZE, .proc/on_paralyze)
+
+	// Shows the radials to the aimer and target
+	src.target.aim_react()
+	show_ui(user, target, stage="start")
+
+/*
+
+Methods to alert the aimer about events, usually to signify that they're complying or trying to pull something.
+
+*/
+
+/datum/component/aiming/proc/on_drop()
+	to_chat(user, "<span class='nicegreen'>[target] has dropped something.</span>")
+
+/datum/component/aiming/proc/on_paralyze()
+	to_chat(user, "<span class='nicegreen'>[target] appears to be surrendering!</span>")
+
+/datum/component/aiming/proc/on_equip()
+	new /obj/effect/temp_visual/aiming/suspect_alert(get_turf(target))
+	to_chat(user, "<span class='userdanger'>[target] has equipped something!</span>")
+	SEND_SOUND(user, 'sound/machines/chime.ogg')
+
+/**
+
+Method to show a radial menu to the person who's aiming, in stages:
+
+AIMING_START means they just recently started aiming
+AIMING_RAISE_HANDS means they selected the "raise your hands above your head" command
+AIMING_DROP_WEAPON means they selected the "drop your weapon" command
+
+*/
+
+/datum/component/aiming/proc/show_ui(mob/user, mob/target, stage)
+	var/list/options = list()
+	var/list/possible_actions = list("cancel", "fire")
+	switch(stage)
+		if("start")
+			possible_actions += "raise_hands"
+			possible_actions += "drop_weapon"
+		if("raise_hands")
+			possible_actions += "drop_to_floor"
+			possible_actions += "face_wall"
+			possible_actions += "raise_hands"
+		if("drop_weapon")
+			possible_actions += "drop_to_floor"
+			possible_actions += "drop_weapon"
+			possible_actions += "raise_hands"
+		if("drop_to_floor")
+			possible_actions += "drop_to_floor"
+		if("face_wall")
+			possible_actions += "face_wall"
+	for(var/option in possible_actions)
+		options[option] = image(icon = 'icons/effects/aiming.dmi', icon_state = option)
+	var/choice = show_radial_menu(user, user, options, require_near = FALSE)
+	act(choice)
+
+/datum/component/aiming/proc/act(choice)
+	if(!user || !target)
+		return //If the aim was cancelled halfway through the process, and the radial didn't close by itself.
+	switch(choice)
+		if("cancel") //first off, are they telling us to stop aiming?
+			stop_aiming()
+			return
+		if("fire")
+			fire()
+			return
+		if("raise_hands")
+			user.say("PUT YOUR HANDS BEHIND YOUR HEAD!")
+		if("drop_weapon")
+			user.say("DROP YOUR WEAPON!")
+		if("face_wall")
+			user.say("TURN AROUND AND FACE THE WALL. SLOWLY.")
+		if("drop_to_floor")
+			user.say("ON THE FLOOR, NOW!")
```

Comment:
forced speech bad.
```suggestion
```

Reply:
This isn't something you actively have to use though, it's more so people can give quick commands if they wouldn't otherwise have time to type something out.

Reply:
Eeeeeeehhhh fine

Reply:
Does this have a Cooldown? And is it mime-proof if they get the "fingergun"? There was an issue on NSV a while back where people spammed the text buttons.

Other than that, glad to see this coming to Bee, it opens up much more opportunity to have an interim where somebody knows you're aiming a weapon at them. Many options for player interaction opening up.

Reply:
Didn't have either of those, but I added them. Dunno about the cooldown length, but I can change that later if it's too much/little.

Reply:
Honestly this always felt kind of weird to me where commands were added or randomly said? Why not just make it "click yell" "click what you want to yell"?

Reply:
that's how it works, there's a button that corresponds to each command. When you click it, you say that command.

Reply:
Oh, I'm dumb then!

Reply:
Instead of force say make use of balloon alerts and display something like "[x] commands you to [y]".
Don't forget to display balloon alert for viewers with appropriate message